README_maintainers.md

```diff
@@ -642,209 +624,261 @@ These commands for `-c` mean:
+
+    ./scripts/unittest find --test tests/js/common/shell/shell-aqlfunctions.js
+
+Run all suite(s) associated with a specific test file:
+
+    ./scripts/unittest auto --test tests/js/common/shell/shell-aqlfunctions.js
+
+Run all tests:
+
+    scripts/unittest all
+
+`scripts/unittest` is only a wrapper for the most part, the backend
+functionality lives in `js/client/modules/@arangodb/testing.js`.
+
+#### Passing Options
+
+The first parameter chooses the facility to execute.
+Available choices include:
+
+ - **all**:                This target is utilized by most of the Jenkins builds invoking unit tests (calls multiple)
+ - **single_client**:      (see [Running a single unittest suite](#running-a-single-unittest-suite))
+ - **single_server**:      (see [Running a single unittest suite](#running-a-single-unittest-suite))
+
+Different facilities may take different options. The above mentioned usage
+output contains the full detail.
+
+A commandline for running a single test (-> with the facility 'single_server') using
+valgrind could look like this. Options are passed as regular long values in the
+syntax --option value --sub:option value. Using Valgrind could look like this:
+
+    ./scripts/unittest single_server --test js/server/tests/aql/aql-escaping.js \
+      --extraArgs:server.threads 1 \
+      --extraArgs:scheduler.threads 1 \
+      --extraArgs:javascript.gc-frequency 1000000 \
+      --extraArgs:javascript.gc-interval 65536 \
+      --extraArgs:log.level debug \
+      --extraArgs:log.force-direct true \
+      --javascript.v8-contexts 2 \
+      --valgrind /usr/bin/valgrind \
+      --valgrindargs:log-file /tmp/valgrindlog.%p
+
+- we specify the test to execute
+- we specify some arangod arguments via --extraArgs which increase the server performance
+- we specify to run using valgrind (this is supported by all facilities)
+- we specify some valgrind commandline arguments
+- we set the log level to debug
+- we force the logging not to happen asynchronous
+- eventually you may still add temporary `console.log()` statements to tests you debug.
+
+#### Running a Single Unittest Suite
+
+Testing a single test with the framework directly on a server:
+
+    scripts/unittest single_server --test tests/js/server/aql/aql-escaping.js
+
+You can also only execute a filtered test case in a jsunity/mocha/gtest test suite (in this case `testTokens`):
+
+    scripts/unittest single_server --test tests/js/server/aql/aql-escaping.js --testCase testTokens
+    
+    scripts/unittest shell_client --test shell-util-spec.js --testCase zip
+    
+    scripts/unittest gtest --testCase IResearchDocumentTest.*
+
+Testing a single test with the framework via arangosh:
+
+    scripts/unittest single_client --test tests/js/server/aql/aql-escaping.js
+
+Testing a single rspec test:
+
+    scripts/unittest http_server --test api-users-spec.rb
+
+Running a test against a ready started server (in contrast to starting one by itself):
```

Comment:
Running a test against a server that you had already started before (in contrast of the test launching the test-server on its own)